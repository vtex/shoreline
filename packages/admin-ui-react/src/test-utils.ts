import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
