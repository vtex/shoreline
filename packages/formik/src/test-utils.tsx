import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
