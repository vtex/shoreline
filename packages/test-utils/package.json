{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
