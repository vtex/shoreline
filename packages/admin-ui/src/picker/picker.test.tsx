import React from 'react'
import { axe, fireEvent, render, screen } from '../test-utils'

import type { PickerInitialState } from './index'
import {
  Picker,
  PickerPopover,
  PickerDisclosure,
  usePickerState,
} from './index'

function BaseComponent(props: PickerInitialState) {
  const state = usePickerState({
    ...props,
    pickerId: 'selector-1',
    popoverId: 'dialog-1',
    baseId: 'picker-test',
  })

  return (
    <>
      <Picker aria-label="picker base" state={state}>
        <PickerDisclosure state={state}>open</PickerDisclosure>
      </Picker>
      <PickerPopover
        aria-label="picker base content"
        data-testid="testid-picker-dialog"
        state={state}
      >
        <button>tabbable</button> Content
      </PickerPopover>
    </>
  )
}

describe('Selector', () => {
  it('should render correctly', () => {
    const { baseElement } = render(<BaseComponent visible />)

    expect(baseElement).toMatchSnapshot()
  })

  it('should open/close properly', () => {
    render(<BaseComponent />)

    const pickerContent = screen.getByTestId('testid-selector-dialog')

    expect(pickerContent).not.toBeVisible()
    fireEvent.click(screen.getByText('open'))
    expect(pickerContent).toBeVisible()
  })

  it('Selector renders with no a11y violations', async () => {
    const { container } = render(<BaseComponent />)
    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
