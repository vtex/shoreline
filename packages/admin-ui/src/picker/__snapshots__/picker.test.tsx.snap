// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker should render correctly 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div>
    <div
      aria-expanded="true"
      aria-haspopup="dialog"
      aria-label="picker base"
      aria-owns="dialog-1"
      class="emotion-0"
      id="selector-1"
      role="button"
    >
      <button
        aria-controls="picker-test"
        aria-expanded="true"
        aria-haspopup="dialog"
        class="emotion-0"
        tabindex="-1"
        type="button"
      >
        open
      </button>
    </div>
  </div>
  <div
    aria-hidden="true"
    class="__reakit-focus-trap"
    style="position: fixed;"
    tabindex="0"
  />
  <div
    class="__reakit-portal"
  >
    <div
      aria-label="picker base content"
      aria-modal="true"
      class="emotion-0"
      data-dialog="true"
      data-testid="testid-picker-dialog"
      id="dialog-1"
      role="dialog"
      style="position: fixed; left: 100%; top: 100%;"
      tabindex="-1"
    >
      <button>
        tabbable
      </button>
       Content
    </div>
  </div>
  <div
    aria-hidden="true"
    class="__reakit-focus-trap"
    style="position: fixed;"
    tabindex="0"
  />
</body>
`;
